Merge staging into development

leahcornelius opened the pull request:
This merges both fix-the-dag and p2p-refactor into development (#20 #15, #14)
- [x] Fix all compiler warnings caused by the new functions (99% naming issues)
- [ ] Write unit tests for new code

leahcornelius committed:
[MULTI] Fix handle stream not pausing during sync

leahcornelius committed:
[P2P][HANDLE] Remove surplus logging

leahcornelius committed:
[P2P][HANDLE] Misseed some ... :)

leahcornelius committed:
[P2P] Implment message type 0x60 (get chain list (ask))

leahcornelius committed:
[P2P][HANDLE] Implement message type 0x45 (get block count ask)

leahcornelius committed:
[P2P][HANDLE] Implement message type 0x6f (get blocks above hash (ask))

leahcornelius committed:
[BLOCKCHAIN] Check read chain height value is not -1 before parsing into u64

leahcornelius committed:
[BLOCKCHAIN] Form_recieve block. use recipitent public key to get top hash, not the senders

leahcornelius committed:
[BLOCKCHAIN] Logging for opening raw block files

leahcornelius committed:
[P2P][HELPER] Check block type before enacting

leahcornelius committed:
[P2P][HANDLE] Check block type before enacting

leahcornelius committed:
[P2P][HELPER] Fix block prop issues

leahcornelius committed:
[BLOCKCHAIN] Update chain_digest and state_digest system

leahcornelius committed:
[CONFIG] Change default host parameter

leahcornelius committed:
create staging branch

leahcornelius committed:
P2p refactor -> staging (#27)

* [P2P] Wrrite launch server wrapper code

* [P2P] != -> ==

* [P2P] P2P server logging

* [P2P] debug level logging fr new incoming dtreams

* [BLOCKCHAIN] Fix the dag structure (#16)

* [BLOCKCHAIN] Add block type to block struct

* [BLOCKCHAIN][BLOCK] Only recieve blocks can be enacted, return a error if we try to enact a send block

* [BLOCKCHAIN] Check if block is already in mempool before enacting

* [BLOCKCHAIN][CARGO] Change my psudoacc name to my real name

* [BLOCKCHAIN] Dont return err if block is in mempool 

My mistake, a block will ALWAYS be in the mempool before enacting

* [BIN] [DAEMON] Move to work with the new block format (AIP #15)

* [BLOCKCHAIN] [GENESIS] Update genesis to work with new block tags

* [BLOCKCHAIN] add form_recieve_block and impl default for the BlockType enum

* [MULTI] fix issues

* [P2P] Increase length of handshake key, aes_gcm now need at least 16 bytes

* [P2P] Nonce size needs to be 12 now...

* [P2P] Fix the new p2p code and the handshaking system

* [README] Add clangs dev headers (lib clang) as build requirements

As well as clang. I somehow missed that. Duh

leocornelius commented:
@TheDevMinerTV if you get a chance can you review just the p2p refactor aspect and ensure the are no concerns on your end?

leocornelius commented:
Feel free to take a look at the dag to restructure one and see if anything looks blatantly wrong, but bare in mind its not done yet so there are some bits that are off.

leahcornelius committed:
[BIN][SEEDNODE] Remove uneeded mutability

leahcornelius committed:
[BIN][SEEDNODE] Fixing more warnings

leahcornelius committed:
[MULTI] Cargo fmt

leahcornelius committed:
[P2P][IO] Fix some warnings

leahcornelius committed:
[BLOCKCHAIN] Fix some warnings

leahcornelius committed:
[CRYPTO] Fix some warnings

leahcornelius committed:
[P2P][HELPER] Fix some warnings

TheDevMinerTV committed:
Delete culmulative_addr_freq.csv

TheDevMinerTV commented on p2p/src/core.rs:
```diff
@@ -0,0 +1,134 @@
+                } else {
+                    if data.message != "ack" {
+                        return Err(
+                            "peer did not understand our message; key derivitation failed".into(),
```
misspelled

TheDevMinerTV commented on p2p/src/core.rs:
```diff
@@ -0,0 +1,134 @@
+
+pub fn close_all() {}
+
+pub fn close(_peer: std::net::SocketAddr) {}
```
These should be implemented, no?

TheDevMinerTV commented on p2p/src/handle.rs:
```diff
@@ -0,0 +1,369 @@
+extern crate rand_os;
+extern crate x25519_dalek;
+
+static MAX_SYNCING_PEERS: u64 = 8;
```
This should be configurable

TheDevMinerTV commented on p2p/src/helper.rs:
```diff
@@ -0,0 +1,645 @@
+
+    if unlock == true {
+        debug!("Releasing lock on peer");
+        //unlock_peer(peer).unwrap();
```
This should probably do something.

leahcornelius replied:
No, i can remove that code. Functions should not unlock the passed peers as that is up to the parent thread.

TheDevMinerTV commented on p2p/src/message_types.rs:
```diff
@@ -1,36 +1,48 @@
+    message_types.insert(0xa3, "Handshake Confirm Key Test");
+
+    // chain digests
+    message_types.insert(0x1b, "Send Chain Digest (ask)"); // - why are there two? Backward combatability? TODO: remove one
```
I found both in the old code, you're free to remove one :)

leahcornelius replied:
Going to leave for now

TheDevMinerTV commented on p2p/src/server.rs:
```diff
@@ -0,0 +1,287 @@
+            self.connections += 1;
+        }
+    }
+    fn deincrement_connections(&mut self) {
```
Should be `decrement_connections`.

leahcornelius replied:
Solved, also i need to call those functions

leocornelius commented:
Awesome thanks ill fix those in a bit, also reminded me of a couple of things p2p wise I need to implement regarding sharing your peerlist

leahcornelius committed:
[BLOCKCHAIN] Update to use new state digest system

leahcornelius committed:
[P2P][HELPER] Update to use new state digest system

leahcornelius committed:
[BIN][*] Update to use new state digest system

leahcornelius committed:
[BLOCKCHAIN] Check for case in which there are no chains

leahcornelius committed:
[BLOCKCHAIN] Raw_lyra takes a string ref...

leahcornelius committed:
[P2P][HELPER] Recalculate the state digest after we finish syncing before comparing it with the expected CD

leahcornelius committed:
[BLOCKCHAIN] Fix cyclic digest issue

leahcornelius committed:
[P2P][HELPER] Recalculate cd for synced chains

leahcornelius committed:
[BLOCKCHAIN] Chain keys are not always 44 chars long, blacklist instead

leahcornelius committed:
[BLOCKCHAIN] Move blacklist checking to vector filling zone

leahcornelius committed:
[BLOCKCHAIN] Handle cases with only one chain digest

leahcornelius committed:
[BLOCKCHAIN] Borrow roots[0].1

leahcornelius committed:
[BLOCKCHAIN] Reorder length checking

leahcornelius committed:
[BLOCKCHAIN] When forming chain digests check if block one is default first

leahcornelius committed:
[BLOCKCHAIN] Fix some issues with rusts ownership system

leahcornelius committed:
[BLOCKCHAIN] Remove let keyword that was turning temp_leaf into a local var

leahcornelius committed:
[BLOCKCHAIN] Always hash the gensis hash as the root ndoe

leahcornelius committed:
[P2P][HELPER] Don't recalculate chain digest for sync chains

leahcornelius committed:
[BIN][*] Dont enact genesis block, enact the recieve blocks

leahcornelius committed:
[BLOCKCHAIN] Dont allow enacting of genesis send blocks

leahcornelius committed:
[BLOCKCHAIN] Set timestamp on rec blocks and genesis blocks

leahcornelius committed:
[BIN][DAEMON] Enact send the generated send blocks

leahcornelius committed:
[BIN][DAEMON] Sign blocks

leahcornelius committed:
[BIN][DAEMON] Readd a bunch of signatures we removed a few commits ago :o:

leahcornelius committed:
[BIN][DAEMON] Make sure to sign the block AFTER hashing it!

leahcornelius committed:
[BLOCKCHAIN] Update nonce in rec blocks

leahcornelius committed:
[BIN][SEEDNODE] Dont need to touch disk ever iter

leahcornelius committed:
[CORE][TRANSACTION] Don't validate nonce if txn is in a rec block

leahcornelius committed:
[BLOCKCHAIN] Dont modify nonce of txn

leahcornelius committed:
[CONFIG] Add discord token param to config

leahcornelius committed:
[CONFIG] Fix type errs

leahcornelius committed:
[BIN][CARGO] Add neded libs to cargo

leahcornelius committed:
[BIN][SEEDNODE] Move to new logger, add start of discord code

leahcornelius committed:
[CONFIG][CARGO] Bump crate version

leahcornelius committed:
[BIN][SEEDNODE] Continue discord bot code

leahcornelius committed:
[CRYPTO] Add public_key_to_address public method

leahcornelius committed:
[BIN][SEEDNODE] SOme more bot stuff (only local blockchain changes monitored atm)

leahcornelius committed:
[BIN][SEEDNODE] Don't run discord code if discord token == default

leahcornelius committed:
[MULTI] Database lock caching

* [DATABASE][CARGO] Add  lazy static, avrio_crypto as deps for database

* [BIN][*] Update Safe exit

* [BIN][*] Handle ctrl c safley

* [BIN][CARGO] Add ctrlc lib dep

* [DATABASE] Add caching code

* [BIN][*] Init cache on daemon startup

(using a 1gb cache for now)

* [DATABASE] Do not hash the path

* [BIN][*] Update to use new db sigs

* [BLOCKCHAIN] Update to use new db sigs

* [CORE][INVITE] Update to use new db sigs

* [CORE][TRANSACTION] Update to use new db sigs

* [CRYPTO] Update to use new db sigs & cargo fmt

* [P2P] Update to use new db sigs

* [DATABASE] Update db function signatures

* [BIN][*] Cargo fix

* [BLOCKCHAIN] Cargo fix

* [DATABASE] Cargo fix

* [GIT] Update .gitignore

* [BIN][*] Update bins

* [DATABASE] cache db locks, reload them, update paralism thread count

* [DATABASE] Update some interfucntion logic

* [BIN] Formating

* [DATABASE][CARGO] Bump rocksdb version

* [DATABASE] Don't hash the key

* [BIN] Update the ascii art

* [BIN] Update ascii art

* [BIN] Add a todo comment

* [DATABASE] Cache peers DB

* [DATABASE] Log the new cache size

* [BLOCKCHAIN] Fix inv location for enact_send

* [BIN] More verbose time for logs

* [BLOCKCHAIN] FIx type change to += from +

* [DATABASE] Cache values as well as dbs when recaching new DBs

* [DATABASE] Wait for recache to finish before proceeding

* [DATABASE] Disable caching values for now
Please see issue for more details

* [DATABASE] Capitalise CACHE_VALUES

* [DATABASE] Fixed open ahshmap not  reading values

* [SEEDNODE] Update the max height calcualtion

* [DATABASE] Start using reload_cache again

However only for the database file locks, not the files themselves

leahcornelius committed:
[BIN] Move into one daemon

Seednodes are now run with the --seednode and --seednode-ip options rather than a seperate bin

leahcornelius committed:
[BIN] Logging

leahcornelius committed:
[DAEMON] Fix logging

leahcornelius committed:
[BIN] Reanme per source log lvl for avrio-daemon

richreggio commented:
So I gave trying to merge the staging branch with the development branch a go and also worked on removing any new compiler warnings that weren't talking about unused code or imports. Also went through and changed most of what clippy would complain about, mainly explicit returns at the end of functions and using &String as parameters when &str would suffice. There were a couple of instances of integral math that always returned 0 but that is obviously not what was intended so i switched it to floating point (was a part of the parallelism code). Also implemented the Display trait in place of a couple of to_string() functions since doing so implements to_string(). I'm still wrapping my head around parts of the code base and learning the finer points of Rust but I really am impressed by how much work must have gone into this.

This should finish? PR #26 

Not really sure, I've only ever worked on personal repo's and these are my first attempts at contributing to an open source project.

I'm not sure how I could add to this PR or if I have to open a new one.

[Fork with changes](https://github.com/richreggio/avrio-rs)

leocornelius commented:
That's absolutely incredible thanks! I haven't been able to work on avrio for the last couple of days but tomorrow i will take a look at all your changes and fix anything i can see. The code base is currently quite chaotic as unfortunately i made a naive mistake of "i don't need to document this, i'll remember it". Luckily i can read my code and understand what it does, but i am going through and writing comments for each bit as i go along. That being said, if you do need me to explain any part of the code or protocol do just ask. I really appreciate the contributions, and if you are interested in helping out more i'm more than happy to answer any questions!

leahcornelius committed:
Merge richreggio's changes into staging (#31)

* [GITHUB] Add ISSUE_TEMPLATE.md

* Update ISSUE_TEMPLATE.md

* Create SECURITY.md

* [CRYPTO] Remove unused cryptonight code

* [MULTI] Remove dependencys to cryptonight

* [CORE] [TXN] Remove cryptonight dependency from txn code

* [MULTI] Fix the cryptonight dependencys

* Update README.md

* [README] Update readme a bit

* [README] Fix table of contents

* [TRAVIS] Try using xenial

There seems to be an issue with installing clang with the current config (hence the build failures)

* [TRAVIS] Install packages manually

* [MULTI] Fix all compiler warnings (#29)

Thanks to @richreggio for the PR!

* [TRAVIS] Send travis build job notifications to discord

* [TRAVIS] Add discord notif script

* merge and clippy refactor

* implemented Display

Co-authored-by: BuildTools <unconfigured@null.spigotmc.org>
Co-authored-by: Richard Bickford <61102263+richreggio@users.noreply.github.com>
Co-authored-by: Richard Bickford <rzacharski05@gmail.com>

leocornelius commented:
merged your changes int staging, will review and change anything needed. I need to do some small changes as pointed out by dev miner before merging this PR, and i want to be 100% sure that the cross chain transactions issue was not introduced by these changes so i will merge this once that issue is fixed. After that the rest of the stuff needing to be done for a stable release (eg double checking block hashes with multiple peers, gracefully handling wrong messages during sync, not sending peers who are syncing messages etc) as well as finishing off the API and moving the wallet external can be done on development

leahcornelius committed:
[P2P][CORE] Check identiy != ours when connecting to a peer

leahcornelius committed:
[DAEMON] Remove uneeded arg

leahcornelius committed:
Merge branch 'staging' of https://github.com/avrio-project/avrio-rs into staging

leahcornelius committed:
[P2P][CORE] Spelling mistake

leahcornelius committed:
[P2P][SERVER] deincrement_connections -> decrement_connections

leahcornelius committed:
[P2P][HANDLER] Return peer list when asked

leahcornelius committed:
[P2P][HANDLE] Add announce peer message type (0x9a)

leahcornelius committed:
Add announce peer message type (0x9a)

leahcornelius committed:
[P2P][HELPER] Implement get peerlist

leahcornelius committed:
[BIN] Use new peerlist signature

leahcornelius committed:
[NODE][MEMPOOL] Remove unused imports

leahcornelius committed:
[P2P][HANDLE] Cleaning up warnings, remove unused improts

leahcornelius committed:
[P2P][HELPER] Remove unused unlock peer code

leahcornelius committed:
[CONFIG] Add max_syncing_peers param

leahcornelius committed:
[CONFIG][CARGO] Bump version

leahcornelius committed:
[CARGO] Use local version of avrio config, rather than the web one

leahcornelius committed:
[P2P][HANDLE] USe config value for max_syncing_peers

leahcornelius committed:
[ID] Merge

leahcornelius committed:
[RPC] Create RPC api code

leahcornelius committed:
[CARGO] Add RPC folder to workspace

leahcornelius committed:
[DAMEON] Remove discord code from daemon

leahcornelius committed:
[BIN] Remove further discord code

leahcornelius committed:
[BIN] Delete unused folder

leahcornelius committed:
[DAEMON] Fix logging, add no-sync option for first seednode

leahcornelius committed:
[DAEMON] Add avrio_p2p to the logging whitelist

leahcornelius committed:
[P2P][SERVER] Server side self connection protection

leahcornelius committed:
[P2P][CORE] Add handshake cancel message

leahcornelius committed:
[P2P][HANDLE] Handle cases where we cant get our peerlist

leahcornelius committed:
[DAEMON] Logging

leahcornelius committed:
[P2P][HANDLE] Fix handle stream lock case

prevent fromr eading after unlock

leahcornelius committed:
[P2P] FORMAT

leahcornelius committed:
[P2P][SERVER] Delay before shutting down stream so peer can read cancel message

leahcornelius committed:
[P2P][HELPER][SYNC] Fix typo from PR

leahcornelius committed:
[DAEMON] Remove duplicate logging whitelist entries

leahcornelius committed:
[BLOCKCHAIN] Add topblockhash to the blacklisted keys

(In the form state digest)

leahcornelius committed:
[BLOCKCHAIN] Dont +1 to block height when forming a recieve block

leahcornelius committed:
[DAEMON] Change some rec block send txn code

leahcornelius committed:
[RPC] Add client side/listener code

leahcornelius committed:
[RPC] Trim read bytes

This removes the extra 0's that come from allocating the buffer

leahcornelius committed:
[WALLET] Begin wallet code

leahcornelius committed:
[BIN][CARGO] Add deps for wallet

leahcornelius committed:
[RPC] Handle errs, publize stuff

leahcornelius committed:
[DAEMON] Update logging

leahcornelius committed:
[WALLET] Update log file name

leahcornelius committed:
[WALLET] Update default RPC  server port

leahcornelius committed:
[CORE][ACCOUNT] Prevent panic in case of input=0 (to_dec)

leahcornelius committed:
[API][CARGO]

leahcornelius committed:
[API] Add new methods

leahcornelius committed:
[BIN][CARGO] Add new deps

leahcornelius committed:
[DAEMON] Add logging whitelist for api

leahcornelius committed:
[WALLET] Add send txn and create chain functionality

leahcornelius committed:
[P2P][SERVER] cargo fmt

leahcornelius committed:
[CRYPTO] Derive clone on Wallet struct

leahcornelius committed:
[RPC] Cargo fmt

leahcornelius committed:
[WALLET] Cargo Fix

leahcornelius committed:
[API] Add new method

Get publickey for username

leahcornelius committed:
[WALLET] Add claim txn command and resolve usernames using API

leahcornelius committed:
[WALLET] Subtract fee from lcoal balance as well

leahcornelius committed:
[WALLET] Shorthand struct init

leahcornelius committed:
[WALLET] Begin username register code

leahcornelius committed:
[WALLET] Add register_username code, and move send_transaction code to function

leahcornelius committed:
[API] Stringify parameters

leahcornelius committed:
[WALLET][BIN] Add rpassword as dep

leahcornelius committed:
[WALLET] Use local function to create rec blocks

leahcornelius committed:
[WALLET] Import default trait not fn

leahcornelius committed:
[RPC] Also announce send blocks submited by API

leahcornelius committed:
[P2P][HANDLE] RPC announce blocks recieved over P2P

leahcornelius committed:
[P2P][CARGO] Add avrio p2p ad dep

leahcornelius committed:
[API] Add new methods

leahcornelius committed:
[WALLET] Remove uneeded string

leahcornelius committed:
[DAEMON] Remove all wallet code from daemon

leahcornelius committed:
[WALLET] Add help message for wallet

leahcornelius committed:
[CORE][TXN] Check if type is supported and add burn type

leahcornelius committed:
[WALLET] Add burn command, check balance before sending txns

leahcornelius committed:
[API] Add new methods

publickey_to_address and username_for_publickey

leahcornelius committed:
[API] Fix reanme param

leahcornelius committed:
[API] Update method of getting address

leahcornelius committed:
[DAMEON] Use config seednode list

leahcornelius committed:
[CONFIG] Update default seednode list

leahcornelius committed:
[BLOCKCHAIN] Dont write to global (/chaindigest) blockcount/topblockhash

leahcornelius committed:
[API] Create wallet API shell

leahcornelius committed:
[CORE][TRANSACTION] Validate that extra is alphanumeric

This prevents injecting code into the blockchain which breaks... stuff

leahcornelius committed:
[WALLET] Client side verifcation of username

leahcornelius committed:
[CORE][ACCOUNT] Fix logging

leahcornelius committed:
[TRANSACTION] Add failue loggign

leahcornelius committed:
[CORE][TRANSACTION] Move update_nonce to a diffrent function

leahcornelius committed:
[BLOCKCHAIN] Update nonce when enacting send blocks

leahcornelius committed:
[BLOCKCHAIN] Add global blockdatas

leahcornelius committed:
[API] Cargo fmt

leahcornelius committed:
[DAEMON] Use sync_in_order()

leahcornelius committed:
[P2P][HANDLE] Handle new methods for sync_in_order()

leahcornelius committed:
[P2P][HELPER] Add new sync in order function

leahcornelius committed:
[P2P][MSG_TYPES] Add new message types for sync in order

leahcornelius committed:
[DAEMON] Cargo fmt

leahcornelius committed:
[P2P][HANDLE] Cargo fmt

leahcornelius committed:
[P2P][HELPER] Cargo fmt

leahcornelius committed:
[DAEMON] CArgo fix

leahcornelius committed:
[CORE][TRANSACTION] Cargo fix

leahcornelius committed:
[P2P][HANDLE] Check for hash == "-1" as well

||

leahcornelius committed:
[P2P][HELPER] "0" -> 0

leahcornelius committed:
[P2P][HELPER] Dont string encode the zero

leahcornelius committed:
[P2P][HELPER] Dont json decode the string

leahcornelius committed:
[P2P][HANDLE] Clone read_msg.msg

leahcornelius committed:
[P2P][HANDLE] Global inventories start from 1

leahcornelius committed:
[P2P][HANDLE] Use the got value

leahcornelius committed:
[P2P][HANDLE] BReak from loop once we read a -1 value

leahcornelius committed:
[WALLET] call to_dec on all atomic balances

leahcornelius committed:
[P2P][HELPER] Increase verbosity, reutrn if there are no blocks

leahcornelius committed:
[RPC] Log new blocks

leahcornelius committed:
[BLOCKCHAIN] Add new block validation code and a tempory way of benchmarking it

leahcornelius committed:
[BLOCKCHAIN] Add new block validation code and a tempory way of benchmarking it

leahcornelius committed:
[P2P][HELPER] Cargo fmt

leahcornelius committed:
[P2P][HELPER] Cargo fmt

leahcornelius committed:
[API] Add new deps, bump rocket

leahcornelius committed:
[API] Add new deps, bump rocket

leahcornelius committed:
[API] Begin moving wallet API

leahcornelius committed:
[API] Begin moving wallet API

leahcornelius committed:
[API] Moving wallet api

leahcornelius committed:
[API] Moving wallet api

leahcornelius committed:
[BIN] Add rocket as dep (for walelt api)

leahcornelius committed:
[BIN] Add rocket as dep (for walelt api)

leahcornelius committed:
[BIN][WALLET] Begin moving wallet api

leahcornelius committed:
[BIN][WALLET] Begin moving wallet api

leahcornelius committed:
[WALLET] Convert atomic units to dec

leahcornelius committed:
[WALLET] Convert atomic units to dec

leahcornelius committed:
[BLOCKCHAIN] Allow error codes of 0 while getting accounts

leahcornelius committed:
[BLOCKCHAIN] Allow error codes of 0 while getting accounts

leahcornelius committed:
[BLOCKCHAIN] Allow parent child age parity

leahcornelius committed:
[BLOCKCHAIN] Allow parent child age parity

leahcornelius committed:
[BLOCKCHAIN] Update benchmark code

leahcornelius committed:
[BLOCKCHAIN] Update benchmark code

leahcornelius committed:
[BLOCKCHAIN] Dont double verify blocks anymore

leahcornelius committed:
[BLOCKCHAIN] Dont double verify blocks anymore

leahcornelius committed:
[BLOCKCHAIN] Fix varible name

leahcornelius committed:
[BLOCKCHAIN] Fix varible name

leahcornelius committed:
[P2P][HELPER] Increase timeout for prop block

leahcornelius committed:
[P2P][HELPER] Increase timeout for prop block

leahcornelius committed:
[P2P][HELPER] Unlock peer when global block count = 0

leahcornelius committed:
[P2P][HELPER] Unlock peer when global block count = 0

leahcornelius committed:
[P2P][IO] Move to new AES GCM code

leahcornelius committed:
[P2P][IO] Move to new AES GCM code

leahcornelius committed:
[P2P][IO]

leahcornelius committed:
[P2P][IO]

leahcornelius committed:
[WALLET] Trim windows shit from strings

leahcornelius committed:
[WALLET] Trim windows shit from strings

leahcornelius committed:
[DATABASE] Change the way we encode the peerlist

leahcornelius committed:
[DATABASE] Change the way we encode the peerlist

leahcornelius committed:
[P2P][CORE] Reduce timeout, dont reconnect to peers

leahcornelius committed:
[P2P][CORE] Reduce timeout, dont reconnect to peers

leahcornelius committed:
[P2P][CORE] Allow a bit more timeout

leahcornelius committed:
[P2P][CORE] Allow a bit more timeout

leahcornelius committed:
[DAEMON] Change state digest regeneration criteria

leahcornelius committed:
[DAEMON] Change state digest regeneration criteria

leahcornelius committed:
[BLOCKCHAIN] Check the txn is contained in the got send block, not the got previous block

leahcornelius committed:
[BLOCKCHAIN] Check the txn is contained in the got send block, not the got previous block

leahcornelius committed:
[CORE] Transaction add new transaction validation code

leahcornelius committed:
[CORE] Transaction add new transaction validation code

leahcornelius committed:
[CORE][GAS] Update gas amounts for message types

leahcornelius committed:
[CORE][GAS] Update gas amounts for message types

leahcornelius committed:
[BLOCKCHAIN] Use new transaction.valid() signature

leahcornelius committed:
[BLOCKCHAIN] Use new transaction.valid() signature

leahcornelius committed:
[CORE][TRANSACTION] if self.extra != "" rather than checking for sizw

leahcornelius committed:
[CORE][TRANSACTION] if self.extra != "" rather than checking for sizw

leahcornelius committed:
[CORE][TRANSACTION] use .len() rather than size of val()

leahcornelius committed:
[CORE][TRANSACTION] use .len() rather than size of val()

leahcornelius committed:
[WALLET]  Update gas calcualtions

leahcornelius committed:
[WALLET]  Update gas calcualtions

leahcornelius committed:
[WALLET] Start allowing including data with the command

leahcornelius committed:
[WALLET] Start allowing including data with the command

leahcornelius committed:
[HELPER] Send Sync Close mesage to peer after done

leahcornelius committed:
[HELPER] Send Sync Close mesage to peer after done

leahcornelius committed:
[P2P][MESSAGE TYPES] Add new message types

leahcornelius committed:
[P2P][MESSAGE TYPES] Add new message types

leahcornelius committed:
[P2P][CORE] Check for new 'resend handshake message'

leahcornelius committed:
[P2P][CORE] Check for new 'resend handshake message'

leahcornelius committed:
[P2P][HANDLE] Add ping/pong and rehandshake

leahcornelius committed:
[P2P][HANDLE] Add ping/pong and rehandshake

leahcornelius committed:
Remove reference to the old cryptonight chain

leahcornelius committed:
[P2P][PEER] Add remove peer method

leahcornelius committed:
[P2P][SERVER] Allow rehandshaking

leahcornelius committed:
Merge branch 'new-block-methods' into staging

leahcornelius committed:
Merge branch 'development' into staging
